src/Components/test/testassets/BasicTestApp/InteropComponent.razor

```diff
@@ -175,6 +176,41 @@
             ReturnValues["roundTripByteArrayWrapperObjectFromDotNet"] = ((IJSInProcessRuntime)JSRuntime).Invoke<ByteArrayInterop.ByteArrayWrapper>("roundTripByteArrayWrapperObject", byteArrayWrapperObject).ToString();
         }
 
+        // Streaming interop tests need to be run only for Blazor server
+        // As such, we set the result of `jsToDotNetStreamAsync` to the expected result for WASM
+        if (shouldSupportSyncInterop)
+        {
+            // In the WASM case, setting the result to the expected result ensures the missing test case doesn't result in test failure.
+            ReturnValues["jsToDotNetStreamAsync"] = "Success";
+        }
+        else
+        {
+            // Proceed to run the actual blazor server streaming interop tests
+            var dataReference = await JSRuntime.InvokeAsync<IJSDataReference>("jsToDotNetStreamAsync");
+            using var stream = await dataReference.OpenReadStreamAsync();
+            using var memoryStream = new System.IO.MemoryStream();
+            await stream.CopyToAsync(memoryStream);
+            var buffer = memoryStream.ToArray();
+
+            for (var i = 0; i < buffer.Length; i++)
+            {
+                var expectedValue = i % 256;
+                if (buffer[i] != expectedValue)
+                {
+                    ReturnValues["jsToDotNetStreamAsync"] = $"Failure at index {i}.";
+                    break;
+                }
+            }
+
+            if (buffer.Length != 100_000)
+            {
+                ReturnValues["jsToDotNetStreamAsync"] = $"Failure, got a stream of length {buffer.Length}, expected a length of 50,000.";
+            }
+
+            // Mark the test as successful if we haven't already set a failure above
+            ReturnValues.TryAdd("jsToDotNetStreamAsync", "Success");
```

Comment:
This E2E test covers the case when an `IJSDataReference` is a return value on a .NET-to-JS call.

Should there also be an E2E test case for when it's a parameter on a JS-to-.NET call?

Reply:
https://github.com/dotnet/aspnetcore/pull/33491/commits/eafd3f8a4c5f7f729e35a388ee6e7f4238c79d61